records_mover/records/targets/data_url.py

```diff
@@ -22,14 +22,18 @@ def __init__(self,
                  output_loc: BaseFileUrl,
                  records_format: Optional[BaseRecordsFormat]) -> None:
         if records_format is None:
-            # if we have been given free rein on format, don't
-            # surprise the user by writing a compression they're not
-            # expecting.
+            records_format = DelimitedRecordsFormat()
+
+        if (isinstance(records_format, DelimitedRecordsFormat) and
+           'compression' not in records_format.custom_hints):
```

Comment:
Now so long as the user didn't specifically set a hint, we'll do inference on the file extension.